### PR TITLE
feat: airdrop ws integrations initial

### PR DESCRIPTION
**Pull request by PanchoBubble**
Websocket integrations, and animation triggers

### CODE REVIEW

File: src-tauri/tauri.conf.json [hunk 1]
```diff
@@ -29,6 +29,9 @@
                 "open": true,
                 "sidecar": false
             },
+          "http": {
+            "all": true
```
Comment (Krakaw) [inline src-tauri/tauri.conf.json:33]: Still need to verify we actually need this.

File: src-tauri/tauri.conf.json [hunk 2]
```diff
@@ -63,7 +66,8 @@
             }
         ],
         "security": {
-            "csp": null
+            "csp": null,
+          "dangerousUseHttpScheme": true
```
Comment (Krakaw) [inline src-tauri/tauri.conf.json:70]: Also this.

File: src/hooks/airdrop/useWebsocket.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+export const useWebsocket = () => {
+    const airdropToken = useAirdropStore((state) => state.airdropTokens?.token);
+    const userId = useAirdropStore((state) => state.userDetails?.user?.id);
+    const setUserGems = useAirdropStore((state) => state.setUserGems);
+    const baseUrl = useAirdropStore((state) => state.backendInMemoryConfig?.airdropApiUrl);
+
+    const init = () => {
+        if (!socket && baseUrl) {
+            socket = io(baseUrl, { secure: true, path: '/socket.io', transports: ['websocket'] });
```
Comment (Krakaw) [inline src/hooks/airdrop/useWebsocket.ts:16]: Is the `/socket.io` path not a default?

File: src/hooks/airdrop/useWebsocket.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                if (msgParsed.data.userPoints?.gems) {
+                    setUserGems(msgParsed.data.userPoints?.gems);
+                }
+            });
+            socket.on(userId as string, (msg: string) => {
+                const msgParsed = JSON.parse(msg) as QuestCompletedEvent;
+                if (msgParsed.data.userPoints?.gems) {
+                    setUserGems(msgParsed.data.userPoints?.gems);
+                }
+            });
```
Comment (Krakaw) [inline src/hooks/airdrop/useWebsocket.ts:35]: Why is this duplicated?
```suggestion
            socket.on(userId as string, (msg: string) => {
                const msgParsed = JSON.parse(msg) as QuestCompletedEvent;
                if (msgParsed.data.userPoints?.gems) {
                    setUserGems(msgParsed.data.userPoints?.gems);
                }
            });
```

**Review by Krakaw — COMMENTED** [review]
Looks good, some small things to check

**Review by Krakaw — COMMENTED** [review]
Need to run a `cargo clippy -- -Dwarnings`  but good to go. Before we merge I'm going to get a final answer on the http-all business

**Comment by Anziskavds** [comment]
To test: once you mine a block, you ern tXTM your gem balance should also increase in less than 30s if all is good. I haven't been able to test this because I haven't won any blocks as of yet on v0.5.31.

**Pull request merged**